test/SignerBondsUniswapV2.test.js

```diff
@@ -0,0 +1,251 @@
+const chai = require("chai")
+const expect = chai.expect
+
+const { deployMockContract } = require("@ethereum-waffle/mock-contract")
+const { to1e18, lastBlockTime } = require("./helpers/contract-test-helpers")
+const CoveragePool = require("../artifacts/contracts/CoveragePool.sol/CoveragePool.json")
+const { BigNumber } = ethers
+
+describe("SignerBondsUniswapV2", () => {
+  let governance
+  let riskManager
+  let other
+  let uniswapV2RouterStub
+  let uniswapV2PairStub
+  let mockCoveragePool
+  let signerBondsUniswapV2
+
+  const assetPoolAddress = "0x6e7278c99ac5314e53a3E95b2343D4C57FD46159"
+  // Real KEEP token mainnet address in order to get a verifiable pair address.
+  const tokenAddress = "0x85Eee30c52B0b379b046Fb0F85F4f3Dc3009aFEC"
+
+  beforeEach(async () => {
+    governance = await ethers.getSigner(0)
+    riskManager = await ethers.getSigner(1)
+    other = await ethers.getSigner(2)
+
+    const UniswapV2RouterStub = await ethers.getContractFactory(
+      "UniswapV2RouterStub"
+    )
+    uniswapV2RouterStub = await UniswapV2RouterStub.deploy()
+    await uniswapV2RouterStub.deployed()
+
+    const UniswapV2PairStub = await ethers.getContractFactory(
+      "UniswapV2PairStub"
+    )
+    uniswapV2PairStub = await UniswapV2PairStub.deploy()
+    await uniswapV2PairStub.deployed()
+
+    mockCoveragePool = await deployMockContract(governance, CoveragePool.abi)
+    await mockCoveragePool.mock.assetPool.returns(assetPoolAddress)
+    await mockCoveragePool.mock.collateralToken.returns(tokenAddress)
+
+    const SignerBondsUniswapV2 = await ethers.getContractFactory(
+      "SignerBondsUniswapV2Stub"
+    )
+    signerBondsUniswapV2 = await SignerBondsUniswapV2.deploy(
+      uniswapV2RouterStub.address,
+      mockCoveragePool.address
+    )
+    await signerBondsUniswapV2.deployed()
+
+    // SignerBondsUniswapV2 has to point to the deployed UniswapV2PairStub
+    // instance to make tests work. However, before setting the new value
+    // via the stub, assert the initial value is correct and the pair
+    // address computing logic works well.
+    expect(await signerBondsUniswapV2.uniswapPair()).to.be.equal(
+      // address of real KEEP/ETH LP pool
+      "0xE6f19dAb7d43317344282F803f8E8d240708174a"
+    )
+    await signerBondsUniswapV2.setUniswapPair(uniswapV2PairStub.address)
```

Comment:
What do you think about mocking `uniswapV2Pair` contract? We can do something like this:
```
    mockUniswapPair = await deployMockContract(governance, IUniswapV2Pair.abi)
    await mockUniswapPair.mock.getReserves.returns(42, 32, 111111) // <- return whatever you want
    await signerBondsUniswapV2.setUniswapPair(mockUniswapPair.address)
```

By mocking ^ we can get rid of `UniswapV2PairStub.sol`

Reply:
In contrary to https://github.com/keep-network/coverage-pools/pull/66#discussion_r644244640, I think replacing `UniswapV2PairStub` with Waffle mock is a good idea, as this stub is very simple and we do some boilerplate here. WIll change it according to your proposition.

Reply:
Please see https://github.com/keep-network/coverage-pools/pull/66/commits/c828b33c022f6043dbd98f812e8a44f7e058b75c